src/modules/tls/tls_mod.h

```diff
@@ -30,6 +30,15 @@
 #include "../../core/locking.h"
 #include "tls_domain.h"
 
+#ifndef OPENSSL_NO_ENGINE
+typedef struct tls_engine {
+        str engine;
+        str engine_config;
+        str engine_algorithms;
+} tls_engine_t;
+#include <openssl/conf.h>
```

Comment:
I think you don't need this includes in the header file, consider move this to the tls_mod.c file.